addons/event/models/event_ticket.py

```diff
@@ -127,14 +127,33 @@ def _constrains_dates_coherency(self):
             if ticket.start_sale_datetime and ticket.end_sale_datetime and ticket.start_sale_datetime > ticket.end_sale_datetime:
                 raise UserError(_('The stop date cannot be earlier than the start date.'))
 
-    @api.constrains('seats_available', 'seats_max')
-    def _constrains_seats_available(self):
-        for record in self:
-            if record.seats_max and record.seats_available < 0:
-                raise ValidationError(
-                    _('No more available seats for the ticket %s (%s). '
-                      'Raise the limit or remove some other confirmed registrations first.',
-                      record.name, record.event_id.name))
+    @api.constrains('registration_ids', 'seats_max')
+    def _check_seats_availability(self):
+        for ticket in self:
+            if ticket.seats_max and ticket.seats_available < 0:
+                raise ValidationError(_(
+                    'Impossible to have a number of confirmed registrations '
+                    'above the seat limit for the ticket %(ticket_name)s '
+                    '(%(event_name)s). Raise the limit or remove some other '
+                    'confirmed registrations first.',
+                    ticket_name=ticket.name, event_name=ticket.event_id.name))
+
+    def name_get(self):
+        """Adds ticket seat availability if requested by context."""
+        if not self.env.context.get('name_with_seat_availability'):
+            return super().name_get()
+        res = []
+        for ticket in self:
+            if ticket.seats_max:
+                if ticket.seats_available:
+                    name = _('%(ticket_name)s (%(seat_count)i seats remaining)',
+                             ticket_name=ticket.name, seat_count=ticket.seats_available)
+                else:
+                    name = _('%(ticket_name)s (sold out)', ticket_name=ticket.name)
+            else:
+                name = _('%(ticket_name)s (unlimited seats)', ticket_name=ticket.name)
```

Comment:
Would the translations work fine if I instead defined a `suffix` with `_()` and then used
```python
res.append((ticket.id, f"{ticket_name} {suffix}"))
```
?

Reply:
Yes, suffix will be translated normally. You may effectively extract "seats remaining".